github/oidc_test.go

```diff
@@ -0,0 +1,20 @@
+package github
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestRequestOIDCToken(t *testing.T) {
+	wantToken := &OIDCToken{JobWorkflowRef: "hoge"}
+	_, stop := NewTestOIDCServer(wantToken)
+	defer stop()
+
+	gotToken, err := RequestOIDCToken("fuga")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if want, got := wantToken, gotToken; !reflect.DeepEqual(want, got) {
```

Comment:
Maybe use github.com/google/go-cmp/cmp, you can then use `cmp.Diff` to show diff.
See example https://github.com/slsa-framework/slsa-github-generator-go/blob/main/pkg/build_test.go